gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -142,34 +157,36 @@ public final ApiClock getClock() {
     return clock;
   }
 
-  // Intended for Internal Use and Overriden by generated ServiceStubSettings classes.
-  // Meant to be shared between StubSettings and ClientContext.
+  /**
+   * Marked with Internal Api and meant to overriden by the generated subclasses. This getter is
+   * used to set the serviceName to the EndpointContext. The value in generated StubSettings
+   * subclasses comes from the proto files.
+   *
+   * <p>This should be effectively treated as an abstract method.
+   */
```

Comment:
We might considered it before, if it is meant to be overridden by sub classes, can we make it `protected`?

Reply:
Ah yep, will update!